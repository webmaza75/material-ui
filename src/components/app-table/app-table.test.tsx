import * as React from 'react';
import * as ShallowRenderer from 'react-test-renderer/shallow';

import AppTable from './app-table';

describe(`AppTable`, () => {
  it(`AppTable renders correctly`, () => {
    const renderer = ShallowRenderer.createRenderer();
    const result = renderer
      .render(<AppTable onItemSelect={jest.fn()} selected={[]} />);

    expect(result).toMatchSnapshot();
  });
});
