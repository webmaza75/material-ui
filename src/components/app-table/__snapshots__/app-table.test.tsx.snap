// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTable AppTable renders correctly 1`] = `
<div
  style={
    Object {
      "maxWidth": "100%",
    }
  }
>
  <ForwardRef(WithStyles)
    columns={
      Array [
        Object {
          "cellStyle": Object {
            "borderBottom": 0,
            "fontSize": 16,
            "paddingBottom": 2,
            "paddingTop": 2,
          },
          "field": "type",
          "headerStyle": Object {
            "borderBottom": 0,
            "fontSize": 16,
          },
          "render": [Function],
          "sorting": false,
          "title": "",
        },
        Object {
          "cellStyle": Object {
            "borderBottom": 0,
            "fontSize": 16,
            "paddingBottom": 2,
            "paddingTop": 2,
          },
          "field": "title",
          "title": "Task Title",
        },
        Object {
          "cellStyle": Object {
            "borderBottom": 0,
            "fontSize": 16,
            "paddingBottom": 2,
            "paddingTop": 2,
          },
          "field": "timeZone",
          "sorting": false,
          "title": "Time Zone",
        },
        Object {
          "cellStyle": Object {
            "borderBottom": 0,
            "fontSize": 16,
            "paddingBottom": 2,
            "paddingTop": 2,
          },
          "field": "reportTime",
          "render": [Function],
          "sorting": false,
          "title": "Report Time",
        },
        Object {
          "cellStyle": Object {
            "borderBottom": 0,
            "fontSize": 16,
            "paddingBottom": 2,
            "paddingTop": 2,
          },
          "field": "repeat",
          "render": [Function],
          "sorting": false,
          "title": "Repeat",
        },
      ]
    }
    components={
      Object {
        "Row": [Function],
      }
    }
    data={
      Array [
        Object {
          "id": 1,
          "repeat": Array [
            6,
            0,
            1,
            2,
            3,
            4,
            5,
          ],
          "reportTime": "Wed Jul 17 2019 10:00:13 GMT+0300",
          "timeZone": "Asia/Aqtu UTC+5:00",
          "title": "Auto Orica USA report setting",
          "type": "Payload Monitoring Report",
        },
        Object {
          "id": 2,
          "repeat": Array [
            0,
            6,
          ],
          "reportTime": "Wed Jul 17 2019 09:00:13 GMT+0300",
          "timeZone": "Asia/Aqtu UTC+5:00",
          "title": "Motion Metrics International",
          "type": "Payload Monitoring Report",
        },
        Object {
          "id": 3,
          "repeat": Array [
            1,
            2,
            3,
            4,
          ],
          "reportTime": "Wed Jul 17 2019 15:00:13 GMT+0300",
          "timeZone": "America/Los_Angeles UTC+8:00",
          "title": "MPM auto weekly report task",
          "type": "Fragmentation Report",
        },
        Object {
          "id": 4,
          "repeat": Array [
            4,
          ],
          "reportTime": "Wed Jul 17 2019 09:30:13 GMT+0300",
          "timeZone": "Africa/Johannesburg UTC+2:00",
          "title": "Orica USA auto report",
          "type": "Tooth Detection Report",
        },
        Object {
          "id": 5,
          "repeat": Array [
            1,
            2,
            4,
            5,
            3,
          ],
          "reportTime": "Wed Jul 17 2019 09:30:13 GMT+0300",
          "timeZone": "Africa/Johannesburg UTC+2:00",
          "title": "Orica USA auto report",
          "type": "Tooth Detection Report",
        },
      ]
    }
    onRowClick={[Function]}
    options={
      Object {
        "headerStyle": Object {
          "fontSize": 16,
        },
        "maxBodyHeight": 400,
        "paging": false,
        "rowStyle": [Function],
        "search": false,
        "sorting": true,
        "toolbar": false,
      }
    }
    style={
      Object {
        "border": 0,
        "boxShadow": "none",
      }
    }
  />
</div>
`;
